contracts/modules/SSVViews.sol

```diff
@@ -49,13 +50,88 @@ contract SSVViews is ISSVViews {
+    function getOperatorById(
+        uint64 operatorId
+    )
+        external
+        view
+        override
+        returns (
+            address owner,
+            uint256 fee,
+            uint32 validatorCount,
+            address whitelistedAddress,
+            bool isPrivate,
+            bool isActive
+        )
+    {
+        ISSVNetworkCore.Operator storage operator = SSVStorage.load().operators[operatorId];
+
+        owner = operator.owner;
+        fee = operator.fee.expand();
+        validatorCount = operator.validatorCount;
+        whitelistedAddress = SSVStorage.load().operatorsWhitelist[operatorId];
+        isPrivate = operator.whitelisted;
+        isActive = operator.snapshot.block != 0;
+    }
+
+    function getWhitelistedOperators(
+        uint64[] calldata operatorIds,
+        address whitelistedAddress
+    ) external view override returns (uint64[] memory whitelistedOperatorIds) {
+        uint256 operatorsLength = operatorIds.length;
+        if (operatorsLength == 0) return whitelistedOperatorIds;
+
+        StorageData storage s = SSVStorage.load();
+
+        // create the max number of masks that will be updated
+        uint256[] memory masks = OperatorLib.generateBlockMasks(operatorIds, false, s);
+
+        uint256 count;
+        whitelistedOperatorIds = new uint64[](operatorsLength);
+
+        uint256 whitelistedMask;
+        uint256 matchedMask;
+
+        // Check whitelisting status for each mask
+        for (uint256 blockIndex; blockIndex < masks.length; ++blockIndex) {
+            // Only check blocks that have operator IDs
+            if (masks[blockIndex] != 0) {
+                whitelistedMask = s.addressWhitelistedForOperators[whitelistedAddress][blockIndex];
+
+                // This will give the matching whitelisted operators
+                matchedMask = whitelistedMask & masks[blockIndex];
+
+                // Now we need to extract operator IDs from matchedMask
+                uint256 blockPointer = blockIndex << 8;
+                for (uint256 bit; bit < 256; ++bit) {
+                    if (matchedMask & (1 << bit) != 0) {
+                        whitelistedOperatorIds[count++] = uint64(blockPointer + bit);
+                        if (count == operatorsLength) {
+                            return whitelistedOperatorIds; // Early termination
+                        }
+                    }
+                }
+            }
+        }
 
-        return (operator.owner, operator.fee.expand(), operator.validatorCount, whitelisted, isPrivate, isActive);
+        // Resize whitelistedOperatorIds to the actual number of whitelisted operators
+        assembly {
+            mstore(whitelistedOperatorIds, count)
+        }
+    }
+
+    function isWhitelistingContract(address contractAddress) external view override returns (bool) {
+        return OperatorLib.isWhitelistingContract(contractAddress);
+    }
+
+    function isAddressWhitelistedInWhitelistingContract(
```

Comment:
The get that uses our contract can use array of operator ids and addresses, is it worth to make this the same or keep it single?

Reply:
@andrew-blox Do you mean `SSVViews.getWhitelistedOperators()`? that's for SSV whitelisting module addresses. To check many operatorIds and addresses in a external whitelisting contract, instead of implementing ourselves (just a loop of `ISSVWhitelistingContract(whitelistingContract).isWhitelisted(addressToCheck, operatorId)` calls, it's better to add a function to the `ISSVWhitelistingContract` interface and let the users implement it. Like:

`isWhitelisted(address[] addresses, uint64[] operatorIds)`

That's in the case we want to allow users to call external whitelisting contracts through us. External contracts can implement their functions to check the whitelisting status of accounts.

WDYT?